components/cards/SessionCard.tsx

```diff
@@ -0,0 +1,271 @@
+import { getScheduleTimeAndLocation, getSessionTime, getSessionTimeAndLocation, truncate } from '../../util/helpers';
+import Row from '../common/Row';
+import Space from '../common/Space';
+import StyledText from '../common/StyledText';
+
+interface SessionCardProps<T> {
+  handlePress: () => void;
+  handleBookMark?: () => void;
+  item: T;
+  screen?: 'home' | 'sessions';
+  variant?: 'card' | 'list';
+}
+
+const { width } = Dimensions.get('window');
+
+/**
+ *
+ * @param handlePress: function that returns nothing
+ * @param handleBookMark: function that bookmarks and returns nothing
+ * @param item: object of type T
+ * @param screen: 'home' | 'sessions', default is 'home'
+ * @param variant: 'card' | 'list'
+ * @returns SessionCard Component
+ */
+
+const SessionCardOnHome = (props: SessionCardProps<Session>) => {
+  const { handlePress, item } = props;
+  const { colors } = useTheme();
+
+  return (
+    <TouchableWithoutFeedback testID="card" onPress={handlePress}>
+      <View style={[styles.card, { backgroundColor: colors.card }]}>
+        <Image
+          source={{ uri: item.session_image || '' }}
+          style={styles.image}
+          contentFit="cover"
+          contentPosition="left"
+        />
+        <Space size={8} />
+        <View style={styles.bottom}>
+          <View style={styles.description}>
+            <StyledText font="bold" numberOfLines={2} style={styles.title}>
+              {truncate(50, item.title)}
+            </StyledText>
+          </View>
+
+          <Space size={12} />
+
+          <StyledText size="sm" font="light">
+            {getSessionTimeAndLocation(item.slug)}
+          </StyledText>
+        </View>
+      </View>
+    </TouchableWithoutFeedback>
+  );
+};
+
+const SessionCardOnSessions = (props: SessionCardProps<SessionForSchedule>) => {
+  const { handlePress, item, handleBookMark } = props;
+  const { colors } = useTheme();
+
+  return (
+    <TouchableWithoutFeedback testID="card-sessions" onPress={handlePress}>
+      <View style={[styles.card, styles.cardTertiary, { backgroundColor: colors.card }]}>
+        {item.session_image !== null ? (
+          <Image
+            source={{
+              uri:
+                item.session_image ||
+                'https://res.cloudinary.com/droidconke/image/upload/v1668083311/prod/upload/sessions/pwt9pnojtmng8bymxkls.png',
+            }}
+            style={styles.image}
+            contentFit="cover"
+            contentPosition="left"
+          />
+        ) : (
+          <Image
+            source={require('../../assets/images/banner.png')}
+            style={styles.image}
+            contentFit="cover"
+            contentPosition="center"
+          />
+        )}
+
+        <Space size={4} />
+
+        <View style={styles.bottomForSessions}>
+          <StyledText size="sm" font="light">
+            {getScheduleTimeAndLocation(item.start_date_time, item.end_date_time, item.rooms[0] as Room)}
```

Comment:
We should properly handle the case where `items.rooms[0]` is `undefined` in `getScheduleTimeAndLocation` instead of using the assertion to silence TypeScript.

Reply:
On it